session_test.go

```diff
@@ -906,7 +907,7 @@ aws_secret_access_key = DefaultSharedCredentialsSecretKey
 			},
 			Description: "credential validation error",
 			ExpectedError: func(err error) bool {
-				return IsAWSErr(err, "AccessDenied", "")
+				return tfawserr.ErrMessageContains(err, "AccessDenied", "")
```

Comment:
Since we're ignoring the message part, this could be `tfawserr. ErrCodeEquals()` also in `"session creation error"` below.

Reply:
Changed (plus in `session.go` and `awsauth.go`).